internal/provider/setup_key_resource.go

```diff
@@ -0,0 +1,348 @@
+				MarkdownDescription: "SetupKey Name",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+			"expires": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "SetupKey Expiration Date",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+			"expiry_seconds": schema.Int32Attribute{
+				Required:            true,
+				MarkdownDescription: "Expiry time in seconds",
+				PlanModifiers:       []planmodifier.Int32{int32planmodifier.RequiresReplace()},
+			},
+			"updated_at": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Creation timestamp",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"last_used": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Last usage time",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"key": schema.StringAttribute{
+				Computed:            true,
+				Sensitive:           true,
+				MarkdownDescription: "Plaintext setup key",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"type": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+			"usage_limit": schema.Int32Attribute{
+				Computed:            true,
+				Optional:            true,
+				MarkdownDescription: "",
+				PlanModifiers:       []planmodifier.Int32{int32planmodifier.UseStateForUnknown()},
+				Default:             int32default.StaticInt32(0),
+			},
+			"used_times": schema.Int32Attribute{
+				Computed:            true,
+				MarkdownDescription: "",
+				PlanModifiers:       []planmodifier.Int32{int32planmodifier.UseStateForUnknown()},
+			},
+			"state": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"auto_groups": schema.ListAttribute{
+				Computed:            true,
+				MarkdownDescription: "",
+				ElementType:         types.StringType,
+				PlanModifiers:       []planmodifier.List{listplanmodifier.UseStateForUnknown()},
+			},
+			"ephemeral": schema.BoolAttribute{
+				Computed:            true,
+				MarkdownDescription: "",
+				PlanModifiers:       []planmodifier.Bool{boolplanmodifier.UseStateForUnknown()},
+			},
+			"allow_extra_dns_labels": schema.BoolAttribute{
+				Computed:            true,
+				MarkdownDescription: "",
+				PlanModifiers:       []planmodifier.Bool{boolplanmodifier.UseStateForUnknown()},
+			},
+			"valid": schema.BoolAttribute{
+				Computed:            true,
+				MarkdownDescription: "",
+				PlanModifiers:       []planmodifier.Bool{boolplanmodifier.UseStateForUnknown()},
+			},
+			"revoked": schema.BoolAttribute{
```

Comment:
Same as `auto_groups`, this is read-only in docs, but actually revokable (not sure if through the same API).